### DISCUSSION TITLE
Hardware interlock

### DISCUSSION THREAD

**Discussion by IanBod** [arkypita/LaserGRBL#1685] [Ideas]
It would be great to have LaserGRBL output to a different USB port when the laser is running. The USB output could then drive a relay which could be used for an interlock to stop a case being opened for safety or even just to turn on a warning light.

An extension of this could be to require an input as well before turning on the laser. The input would only be present if the case was safely closed.

**Comment by arkypita** [comment]
Usually these things are handled at the hardware level directly from the control board, most controller board have headers just for these switches.

The advantage is that this works regardless of which software you go to use and they also work even if your PC crashes.
The security systems must ALWAYS be independent from the PC, for safety.

**Reply by IanBod** [reply]
As much as anything I was thinking of a light to show when the burn has completed. Like an external version of the green play / red stop button in the software.

Thanks for the quick reply and the great software.

**Comment by arkypita** [comment]
![image](https://user-images.githubusercontent.com/8782035/147514132-7f1b0996-de90-4adb-ad72-d74c76161b7c.png)

Pin A3, coolant enable, usually used to control an additional fan or liquid coolant. Can be turned on/off with M7/M9 command. This is all you need.

**Comment by arkypita** [comment]
P.s if you need job end notification get a look to LaserGRBL telegram bot https://lasergrbl.com/telegram
